doc/manual/src/command-ref/conf-file-prefix.md

```diff
@@ -4,49 +4,57 @@
+    If unset, `XDG_CONFIG_DIRS` defaults to `/etc/xdg`, and `XDG_CONFIG_HOME` defaults to `$HOME/.config` as per [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).
+
+  - If [`NIX_CONFIG`](./env-common.md#env-NIX_CONFIG) is set, its contents are treated as the contents of a configuration file.
+
+### File format
+
+Configuration files consist of `name = value` pairs, one per line.
+Comments start with a `#` character.
+
+Example:
 
-  - If `NIX_USER_CONF_FILES` is set, then each path separated by `:`
-    will be loaded in reverse order.
+```
+keep-outputs = true       # Nice for developers
+keep-derivations = true   # Idem
+```
 
-    Otherwise it will look for `nix/nix.conf` files in `XDG_CONFIG_DIRS`
-    and `XDG_CONFIG_HOME`. If unset, `XDG_CONFIG_DIRS` defaults to
-    `/etc/xdg`, and `XDG_CONFIG_HOME` defaults to `$HOME/.config`
-    as per [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).
+Other files can be included with a line like `include <path>`, where `<path>` is interpreted relative to the current configuration file.
+A missing file is an error unless `!include` is used instead.
 
-  - If `NIX_CONFIG` is set, its contents is treated as the contents of
-    a configuration file.
+A configuration setting usually overrides any previous value.
+However, you can prefix the name of the setting by `extra-` to *append* to the previous value.
 
-The configuration files consist of `name = value` pairs, one per
-line. Other files can be included with a line like `include path`,
-where *path* is interpreted relative to the current conf file and a
-missing file is an error unless `!include` is used instead. Comments
-start with a `#` character. Here is an example configuration file:
+For instance,
 
-    keep-outputs = true       # Nice for developers
-    keep-derivations = true   # Idem
+```
+substituters = a b
+extra-substituters = c d
+```
 
-You can override settings on the command line using the `--option`
-flag, e.g. `--option keep-outputs false`. Every configuration setting
-also has a corresponding command line flag, e.g. `--max-jobs 16`; for
-Boolean settings, there are two flags to enable or disable the setting
-(e.g. `--keep-failed` and `--no-keep-failed`).
+defines the `substituters` setting to be `a b c d`.
 
-A configuration setting usually overrides any previous value. However,
-you can prefix the name of the setting by `extra-` to *append* to the
-previous value. For instance,
+## Command line flags
 
-    substituters = a b
-    extra-substituters = c d
+Every configuration setting has a corresponding command line flag (e.g. `--max-jobs 16`).
+Boolean settings do not need an argument, and can be explicitly disabled with the `no-` prefix (e.g. `--keep-failed` and `--no-keep-failed`).
```

Comment:
```suggestion

There are two ways to set options on the CLI, with a generic `--option` flag, and with option-specific flags.

The generic `--option` way takes two arguments: `--option <name> <value>`.
It is interpreted exactly like a `<name> = <value>` in a setting file.
(In particular `extra-` is supported and unknown options names are warnings.)

Every configuration setting has corresponding command line flag (e.g. `--max-jobs 16`).
Boolean settings do not need an argument, and can be explicitly disabled with the `no-` prefix (e.g. `--keep-failed` and `--no-keep-failed`).
Unknown option names are invalid flags (unless there is already a flag with that name) so unlike with `--option` those become errors not warning.
```

Reply:
Alright, I'll think about how to put this cleanly.